wandb/sdk/artifacts/artifact.py

```diff
@@ -293,67 +289,110 @@ def _from_attrs(
         # Placeholder is required to skip validation.
         artifact = cls("placeholder", type="placeholder")
         artifact._client = client
-        artifact._id = attrs["id"]
         artifact._entity = entity
         artifact._project = project
         artifact._name = name
+        artifact._assign_attrs(attrs)
+
+        artifact.finalize()
+
+        # Cache.
+        assert artifact.id is not None
+        artifact_instance_cache[artifact.id] = artifact
+        return artifact
+
+    def _assign_attrs(self, attrs: dict[str, Any]) -> None:
+        """Populate or repopulate the internal attributes of this Artifact from the parsed, GQL-returned JSON object."""
+        # TODO: In the future, consider dataclasses or generated pydantic types, which can handle much of this.
+        #   Note: pydantic, if considered, readily handles runtime validation logic and camelCase aliases on each field.
```

Comment:
could you limit line length here as well?

<img width="1728" alt="Screenshot 2024-10-16 at 12 29 17 PM" src="https://github.com/user-attachments/assets/eae11fea-0e80-47d1-8612-051e59a39695">

(we've discussed this before---we can't enable a comment line length lint without fixing all existing comments, unfortunately)

(consider readers who use a laptop and split the screen, or people with larger font sizes---80 chars works for almost everyone and is easier to read)

Reply:
> could you limit line length here as well?
> 
> (we've discussed this before---we can't enable a comment line length lint without fixing all existing comments, unfortunately)
>
> (consider readers who use a laptop and split the screen, or people with larger font sizes---80 chars works for almost everyone and is easier to read)

for this specific comment: I think I'll actually delete it altogether, since it's not doing much here.  (even creating a backlog ticket would at least track the suggestion)

for the line limit in general: this has probably been considered (or we discussed and I'm forgetting, in which case 😳), but maybe we can follow up with a single PR to fix the existing comments (or at least most of them) in one go.  That way you don't have to -- justifiably! -- remind me every time I inadvertently tweak or reset my IDE settings :)